src/AutocompleteBadges.js

```diff
@@ -88,6 +162,16 @@ class AutocompleteBadges extends Component {
 
 		this.refs.input.value = '';
 	}
+
+	/**
+	 * Setter for dataItems
+	 * @param val
+	 * @returns {!Array<!Object>}
+	 */
+	setDataItems(val) {
```

Comment:
This method name makes it seem like anyone can call this function on an AutocompleteBadges instance to set `dataItems`, but that will actually just return a copy of the value. That's why we usually name these differently, usually something like `setterDataItems_`, and also make them protected functions (`@protected in the doc block`, so people can know it's not part of the component's public api.

Also, it would be good to explain why this setter is important in this method's doc description, just saying that this is to avoid changing the given array so that it can be passed to different instances for example.

Could you make these minor changes to this method?